packages/next/src/server/dev/hot-reloader-webpack.ts

```diff
@@ -634,7 +634,7 @@ export default class HotReloaderWebpack implements NextJsHotReloaderInterface {
     return span
       .traceChild('clean')
       .traceAsyncFn(() =>
-        recursiveDelete(join(this.dir, this.config.distDir), /^cache/)
+        recursiveDelete(join(this.dir, this.config.distDir), /^(cache|lock)/)
```

Comment:
not exactly sure how this regex is used, but should this be more specific since the lock file is in the root of the distdir?

i forsee a pages dir file called lock.js getting preserved oddly